R/datatype.R

```diff
@@ -85,12 +85,20 @@ print.RPolarsDataType = function(x, ...) {
 "!=.RPolarsDataType" = function(e1, e2) e1$ne(e2)
 
 
-#' check if x is a valid RPolarsDataType
-#' @name is_polars_dtype
-#' @noRd
-#' @param x a candidate
-#' @return a list DataType with an inner DataType
-#' @examples .pr$env$is_polars_dtype(pl$Int64)
+#' Check if input is a valid DataType
+#'
+#' @param x An object to be tested.
+#' @param include_unknown If `FALSE` (default), `pl$Unknown` is considered as
+#' an invalid datatype.
+#'
+#' @export
+#' @return A boolean scalar.
```

Comment:
Is the expression "A boolean scalar" common?

Reply:
No, I just introduced it when working on the Series documentation to differentiate with "A numeric scalar". I can replace it with "A boolean" or just "Boolean", but then I guess I should also replace "A numeric scalar" by something?

Reply:
I think so. I am sure it is not just a problem here.......

Reply:
In general, I think these are "logical", not "boolean".

Reply:
`?all` and `?any` say:

> The value is a logical vector of length one.

But that's very verbose IMO

Reply:
Pls use this:
https://github.com/pola-rs/r-polars/blob/4307c837b6f02022d550d7525f96fe58d49e95b0/R/is_polars.R#L5

Reply:
And move this function to `is_polars.R`

Reply:
So my point is that neither "scalar" nor "boolean" are formal R terms and should not be used in the documentation.

Reply:
I did the replacement for "scalar", but except in some internal function names (`is_numeric_of_length_1()` would be very verbose)

Reply:
Thanks, looks good.

Reply:
For "boolean", it's a bit trickier in some places because the datatype is also named "Boolean", so depending on the context we can prefer "logical" or "boolean". I made some replacements where it's clear we want "logical" but I leave the rest for later.